core/schema/secret.go

```diff
@@ -94,6 +102,51 @@ func (s *secretSchema) setSecret(ctx context.Context, parent *core.Query, args s
 	return i, nil
 }
 
+type newSecretArgs struct {
+	URI string
+}
+
+func (s *secretSchema) newSecret(ctx context.Context, parent *core.Query, args newSecretArgs) (i dagql.Instance[*core.Secret], err error) {
+	clientMetadata, err := engine.ClientMetadataFromContext(ctx)
+	if err != nil {
+		return i, fmt.Errorf("failed to get client metadata from context: %w", err)
+	}
+
+	secretStore, err := parent.Secrets(ctx)
+	if err != nil {
+		return i, fmt.Errorf("failed to get secret store: %w", err)
+	}
+
+	accessor, err := core.GetClientResourceAccessor(ctx, parent, args.URI)
+	if err != nil {
+		return i, fmt.Errorf("failed to get client resource name: %w", err)
+	}
+
+	// NB: to avoid putting the plaintext value in the graph, return a freshly
```

Comment:
@sipsma Do we actually need this, since the secret doesn't container sensitive data?

Reply:
If we support a `plaintext://` url, then yes, because this would be considered sensitive.

Reply:
We were thinking with @sipsma of calling this `plaintext-insecure://` or something, specifically for this reason 😝 

Secret references shouldn't be sensitive, otherwise it implies they have some state somewhere which makes caching impossible.

With this scenario:

- Func A: `return dag.NewSecret("foo")`
- Func B: `A().PlainText()`

If `A()` is function cached and not called, `NewSecret` is not called, but `PlainText` should still work

I think this is not the case right now in this PR since we're carrying data around in the store

Reply:
Ahhh I see. Yeah, that makes sense, I think as long as we make sure that it's noted as insecure it should be good :)

Reply:
I ended up partially addressing this here: https://github.com/dagger/dagger/pull/8730#discussion_r1920920517

But basically I'd just leave it as is for now. The other current utility of it is making the returned ID pure + using a scoped cache key. After my other PR is in that will be way easier but not a problem to leave this alone for the moment.